query.go

```diff
@@ -297,20 +297,19 @@ func (q *query) queryPeer(ctx context.Context, p peer.ID) *queryResult {
 		logger.Debugf("QUERY worker for: %v - not found, and no closer peers.", p)
 	}
 
+	foundCloserPeer := false
 	for _, next := range newPeers {
 		if next.ID == q.dht.self { // don't add self.
 			logger.Debugf("PEERS CLOSER -- worker for: %v found self", p)
 			continue
 		}
 
 		// add their addresses to the dialer's peerstore
-		q.dht.peerstore.AddAddrs(next.ID, next.Addrs, pstore.TempAddrTTL)
-	}
-
-	foundCloserPeer := false
-	for _, np := range newPeers {
-		closer := q.localPeers.Add(np.ID)
-		foundCloserPeer = foundCloserPeer || closer
+		if q.dht.queryPeerFilter(q.dht.host, *next) {
```

Comment:
How necessary is the query filter really, it seems like it's kind of "plan B" since hopefully the routing tables should be clean and only contain relevant peers. Is it ok for functions to overfilter peers here?

Do we also want to filter which addresses we add into the peerstore and/or which ones we eventually dial? If so is it important to do this now?

Reply:
> How necessary is the query filter really, it seems like it's kind of "plan B" since hopefully the routing tables should be clean and only contain relevant peers. Is it ok for functions to overfilter peers here?

For the public DHT, I'd introduce two checks:

1. I can dial the peer.
2. They aren't advertising relay addresses. That's a bit over-zealous, but we don't want to accidentally dial it.

For the private DHT, I'm not sure if we want to filter at all.

> Do we also want to filter which addresses we add into the peerstore and/or which ones we eventually dial? If so is it important to do this now?

I'd actually keep all addresses. If the peer is on the local network, we might as well use those addresses.

Reply:
> I can dial the peer

Sorry, I meant to say "the peer has a public address". Really, just checking for "no relay address" should be sufficient.